Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp

```diff
@@ -981,38 +982,50 @@ static RetainPtr<CTFontDescriptorRef> fontDescriptorWithFamilySpecialCase(const
     if (systemDesign) {
         auto cascadeList = SystemFontDatabaseCoreText::forCurrentThread().cascadeList(fontDescription, family, *systemDesign, allowUserInstalledFonts);
         if (cascadeList.isEmpty())
-            return nullptr;
-        return cascadeList[0];
+            return std::nullopt;
+        return { RetainPtr { cascadeList[0] } };
     }
 
     if (family.startsWith("UICTFontTextStyle"_s)) {
         const auto& request = fontDescription.fontSelectionRequest();
         CTFontSymbolicTraits traits = (isFontWeightBold(request.weight) ? kCTFontTraitBold : 0) | (isItalic(request.slope) ? kCTFontTraitItalic : 0);
         auto descriptor = adoptCF(CTFontDescriptorCreateWithTextStyle(family.string().createCFString().get(), contentSizeCategory(), fontDescription.computedLocale().string().createCFString().get()));
-        if (traits)
-            descriptor = adoptCF(CTFontDescriptorCreateCopyWithSymbolicTraits(descriptor.get(), traits, traits));
-        return descriptor;
+        if (traits) {
+            // FIXME: rdar://105369379 As far as I can tell, there's no modification to the attributes dictionary that has the same effect as CTFontDescriptorCreateCopyWithSymbolicTraits(),
+            // because there doesn't seem to be a place to specify the bitmask.
```

Comment:
Can you mention in the comment that it's for that reason we create the CTFont eagerly?